_posts/2016-06-30-state-in-scalable-architectures.md

```diff
@@ -0,0 +1,173 @@
+---
+layout: post
+name: State-in-scalable-architectures
+title: 'State in scalable architectures'
+date: 2016-06-30 00:20:00 +00:00
+author: Felipe Fernández
+image:
+    src: /assets/img/custom/blog/state.jpg
+    attribution:
+      text: Untitled by Daida Medina
+      href: https://www.instagram.com/p/8TdviFHo-g/?taken-by=thelastfreeusername
+tags:
+- state
+- scalability
+- architecture
+
+---
+
+Handling state is one of the cornerstone of software development. Most of business value derived from software relies on state. Depending on the level of abstraction, state, and the approach to handle it, looks like a complete different problem. In this post we'll see what is common to all of them and how modern scalability needs have pushed the industry to innovative and refreshing approaches.
+
+## Levels of abstraction
+
+[Ted Malaska's presentation](https://www.youtube.com/watch?v=sD-rvRkK_u4) about [Spark](http://spark.apache.org/) serves as inspiration for this post. In this screenshot, Ted shows a list of concepts that we have to understand when learning to code:
+
+<img src="/assets/img/custom/blog/learning-to-code.png" alt="Learning to code" title="Learning to code" class="img img-center img-responsive style-screengrab">
+
+You can stop reading here and spend some time thinking in your conceptions about state at any of those levels. When you're done, come back here and let's try to define more formally what state is.
+
+## State definition
+
+If you went through a similar train of thought like myself these concepts appeared for sure: mutability, concurrency, isolation and scope. Most of them has to do with the level of abstraction or the strategy to handle state, not with the definition itself. Let's try to extract the essential from [Wikipedia's definition](https://www.wikiwand.com/en/State_(computer_science)):
+
+>>The state of a digital logic circuit or computer program is a technical term for all the stored information, at a given instant in time, to which the circuit or program has access.
+
+Data at a given instant in time. Neither behaviour, nor abstractions around it, but pure data.
+
+[Out of the tar pit paper](https://github.com/papers-we-love/papers-we-love/blob/master/design/out-of-the-tar-pit.pdf) has powerful insights about state from a more radical concept than scalability: complexity.
+
+>> One final point is that the type of complexity we are discussing in this paper is that which makes large systems hard to understand. It is this that causes us to expend huge resources in creating and maintaining such systems. This type of complexity has nothing to do with complexity theory — the branch of computer science which studies the resources consumed by a machine executing a program.
+
+## State at service level
+
+Through this post we'll see how simplicity and scalability aligns blissfully when handling state. Let's start with approaches to handle state at service level. We'll move then into distributed systems integration.
```

Comment:
We'll move then into distributed systems integration => We'll then move into distributed systems integration